### PR TITLE
Support atomic ordering for LLVM load/store instructions

### PR DESCRIPTION
**Pull request by AlexeySotkin**
LLVM IR `store atomic` and `load atomic` instructions are now lowered
to the corresponding SPIR-V atomics.

Signed-off-by: Andrew Savonichev <andrew.savonichev@intel.com>
Signed-off-by: Alexey Sotkin <alexey.sotkin@intel.com>

### CODE REVIEW

File: test/atomic-load-store.ll [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  %0 = alloca i32
+
+; CHECK: AtomicStore [[#PTR]] [[#DeviceScope]] [[#Relaxed]] [[#]]
+  store atomic i32 0, i32* %0 monotonic, align 4
+; CHECK: AtomicStore [[#PTR]] [[#DeviceScope]] [[#Release]] [[#]]
+  store atomic i32 0, i32* %0 release, align 4
+; CHECK: AtomicStore [[#PTR]] [[#DeviceScope]] [[#SequentiallyConsistent]] [[#]]
+  store atomic i32 0, i32* %0 seq_cst, align 4
+
+; CHECK: AtomicLoad 6 {{[0-9]+}} [[#PTR]] [[#DeviceScope]] [[#Relaxed]]
```
Comment (MrSidims) [inline test/atomic-load-store.ll:28]: I suggest to either mask result type with `{{[0-9]+}}` (as it was done for result id) or actually check it.

File: test/atomic-load-store.ll [hunk 2]
```diff
@@ -0,0 +1,35 @@
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv -to-text %t.spv -o - | FileCheck %s
```
Comment (MrSidims) [inline test/atomic-load-store.ll:4]: I see, that purpose of this patch is to add a missing functionality in forward translation, but does it make sense to also add a test for a reversed translation to double check it?

**Review by MrSidims — COMMENTED** [review]
Functionally LGTM, just few test related questions.

File: test/atomic-load-store.ll (hunk 2 above)
Comment (AlexeySotkin) [reply test/atomic-load-store.ll:4]: The reverse translation is covered by [test/transcoding/atomics.spt](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/test/transcoding/atomics.spt#L44-L45)

File: test/atomic-load-store.ll (hunk 1 above)
Comment (AlexeySotkin) [reply test/atomic-load-store.ll]: Fixed in 2fbad5e

File: test/atomic-load-store.ll (hunk 2 above)
Comment (AlexeySotkin) [reply test/atomic-load-store.ll:4]: Though test/transcoding/atomics.spt can be extended with --spirv-target-env="SPV-IR", which I have done in 4104901

**Review by MrSidims — APPROVED** [review]

**Pull request merged**